lib/rules/no-implicit-coercion.js

```diff
@@ -106,14 +106,34 @@ function getNonNumericOperand(node) {
 }
 
 /**
+ * Checks whether a node is a string literal or not.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} Whether or not the passed in node is a
+ * string literal or not.
+ */
+function isString(node) {
+    return node.type === "Literal" && typeof node.value === "string";
```

Comment:
Consider using [`isStringLiteral` from ast-utils](https://github.com/eslint/eslint/blob/master/lib/ast-utils.js#L229)? (One significant difference, that function also checks for TemplateLiterals. I think that might be okay here, though.)